### DISCUSSION TITLE
Add 'whereWeek' to Eloquent Query Builder

### DISCUSSION THREAD

**Discussion by tarekharms** [laravel/framework#38504] [Ideas]
Hello there,

the Eloquent Query Builder offers methods like "whereYear" or "whereMonth". What I am missing is a method for selecting by calendarweek like this:   

https://www.w3resource.com/mysql/date-and-time-functions/mysql-week-function.php   

An additional mode selection as described for different time locals would be necessary too.  
I know about the workaround "Sorting by start and end date of a calendar week" but this isn't realy satisfying because of the surpassing of given MySQL functions.

Greetings.

**Comment by muditsunarctech** [comment]
I was searching for the same.
I've created custom builder classes for the same, but it would be great if we have something out of the box.

**Comment by DevKemal** [comment]
It's really weird to have whereYear, whereMonth but not whereWeek, still doesn't seem to work in laravel 9. Here's an alternative : 

```
$query->whereRaw('WEEK(created_at) = ' . date('W'))
```
